### DISCUSSION TITLE
Cors error in React + Node

### DISCUSSION THREAD

**Discussion by ricopomps** [vercel/community#4094] [Help]
### Summary

I deployed a react and a node project in vercel, both work separately, but when the front calls the api it gets a CORS error, i have the same node project deployed in heroku, and if I set the react app to call the heroku api it does not have any problem with cors, so I assume the code is right and the error is in my vercel.json file

### Steps to Reproduce

The vercel file is this: 
```
{
  "version": 2,
  "builds": [
    {
      "src": "src/server.ts",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "(.*)",
      "dest": "src/server.ts"
    }
  ]
}
```

I tried adding this code from the documentation to my vercel.json
```
"headers": [
    {
      "source": "/api/(.*)",
      "headers": [
        { "key": "Access-Control-Allow-Credentials", "value": "true" },
        { "key": "Access-Control-Allow-Origin", "value": "*" },
        { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
        { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
      ]
    }
  ]
```
But the deploy did not work, tried setting the header inside the route but also did not work, how am I supposed to configure cors?

**Comment by rahulpandey70** [comment]
> ### Summary
> I deployed a react and a node project in vercel, both work separately, but when the front calls the api it gets a CORS error, i have the same node project deployed in heroku, and if I set the react app to call the heroku api it does not have any problem with cors, so I assume the code is right and the error is in my vercel.json file
> 
> ### Example
> _No response_
> 
> ### Steps to Reproduce
> The vercel file is this:
> 
> ```
> {
>   "version": 2,
>   "builds": [
>     {
>       "src": "src/server.ts",
>       "use": "@vercel/node"
>     }
>   ],
>   "routes": [
>     {
>       "src": "(.*)",
>       "dest": "src/server.ts"
>     }
>   ]
> }
> ```
> 
> I tried adding this code from the documentation to my vercel.json
> 
> ```
> "headers": [
>     {
>       "source": "/api/(.*)",
>       "headers": [
>         { "key": "Access-Control-Allow-Credentials", "value": "true" },
>         { "key": "Access-Control-Allow-Origin", "value": "*" },
>         { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
>         { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
>       ]
>     }
>   ]
> ```
> 
> But the deploy did not work, tried setting the header inside the route but also did not work, how am I supposed to configure cors?

I think you should install cors and add to your backend `index` or `servar` file like this

`const cors = require('cors')`
 `Const app = express()`
 `app.use(cors()) `
